### ISSUE TITLE
Bias-Variance Tradeoff

### ISSUE THREAD

**Issue by fuleying** [fuleying/fuleying.github.io#1]
![image](https://user-images.githubusercontent.com/6255387/42885333-4b4aae5a-8ad3-11e8-966a-637a38d65382.png)
![image](https://user-images.githubusercontent.com/6255387/42885361-5b58a0e0-8ad3-11e8-85af-06e32c3719df.png)
- 训练集（training set）是我们用来训练模型的
- 开发集（dev set）指的是在开发过程中用于调参,验证（validation）等步骤的数据集（保证不被模型提前学习到）
- 测试集（test set）很显然就是指测试时所使用的数据集。
- 有了这三个数据集，我们就可以得出三个误差值（分别为）：
![image](https://user-images.githubusercontent.com/6255387/42885391-6423f5f8-8ad3-11e8-8063-255d501b3071.png)
- 人类水平的误差（human level error）是人类自己处理这类问题的误差值；
- 训练集误差（training set error）是指在训练集上跑出来的误差值；
- 开发集误差（dev set error）是指用开发集跑出来的误差值。
- 我们关注的并不是这个误差值本身，而是它们**互相之间的差距**。
> - 人类误差与训练集误差之间的差距称为`avoidable bias（可避免的偏差，可简称为偏差）`。之所以说`可避免的（avoidable）`，是因为这部分误差可以通过进一步的学习/模型调整优化来避免。
> - 而训练集和开发集之间的差距称为`variance（方差）`，它是因为跑了不同的数据而导致的误差率变化（比如跑在见过的数据集上和没见过的数据集上的误差率之差）。这两种偏差合在一起，就是机器学习领域著名的 **bias-variance trade-off（偏差-方差权衡）**。

**Comment by fuleying** [comment]
同样是training/dev/test set这三组数据，如果你遇到的情况是：
![image](https://user-images.githubusercontent.com/6255387/42885797-8b785c4c-8ad4-11e8-9079-443e2fd60476.png)
- 左边的情况是：训练集误差率与人类自己的误差率只相差 1%，然而训练集误差跟测试集误差却差了很多；这就意味着你的模型在新的（没见过）的数据上表现很不好，换句话说，你的模型过拟合（overfitting）了。
- 对于右边的情况，如果你的训练集误差跟人类误差值相比就已经差了很多，而测试集误差则更加地多，那么，洗洗睡吧，这模型没戏。

**Comment by fuleying** [comment]
### 机器学习的基本配方：
- 如果训练误差高（偏差），就使用更大的模型、训练更长的时间、采用新的模型架构；
- 如果开发误差高（方差），就使用更多数据、正则化、新的模型架构。
![image](https://user-images.githubusercontent.com/6255387/42886021-1965b072-8ad5-11e8-90c2-d8b948ac4f01.png)
